stock_operating_unit/model/stock.py

```diff
@@ -0,0 +1,182 @@
+    _inherit = "stock.warehouse"
+
+    def _default_operating_unit(self):
+        if self.company_id:
+            company = self.company_id
+        else:
+            company = self.env['res.company']._company_default_get(
+                'stock.inventory')
+        for ou in self.env.user.operating_unit_ids:
+            if company == self.company_id:
+                self.operating_unit_id = ou
+
+    operating_unit_id = fields.Many2one(
+        comodel_name='operating.unit',
+        string='Operating Unit',
+        default=_default_operating_unit
+    )
+
+    @api.multi
+    @api.constrains('operating_unit_id', 'company_id')
+    def _check_company_operating_unit(self):
+        for rec in self:
+            if (rec.company_id and rec.operating_unit_id and
+                    rec.company_id != rec.operating_unit_id.company_id):
+                raise UserError(
+                    _('Configuration error. The Company in the Stock Warehouse'
+                      ' and in the Operating Unit must be the same.')
+                )
+
+
+class StockLocation(models.Model):
+    _inherit = 'stock.location'
+
+    operating_unit_id = fields.Many2one('operating.unit', 'Operating Unit')
+
+    @api.multi
+    @api.constrains('operating_unit_id')
+    def _check_warehouse_operating_unit(self):
+        for rec in self:
+            warehouse_obj = self.env['stock.warehouse']
+            warehouses = warehouse_obj.search(
+                ['|', '|', ('wh_input_stock_loc_id', '=', rec.ids[0]),
+                 ('lot_stock_id', 'in', rec.ids),
+                 ('wh_output_stock_loc_id', 'in', rec.ids)])
+            for w in warehouses:
+                if rec.operating_unit_id != w.operating_unit_id:
+                    raise UserError(_('Configuration error. This location is '
+                                      'assigned to a warehouse that belongs to'
+                                      ' a different operating unit.'))
+                if rec.operating_unit_id != w.operating_unit_id:
+                    raise UserError(_('Configuration error. This location is '
+                                      'assigned to a warehouse that belongs to'
+                                      ' a different operating unit.'))
+                if rec.operating_unit_id != w.operating_unit_id:
+                    raise UserError(_('Configuration error. This location is'
+                                      ' assigned to a warehouse that belongs'
+                                      ' to a different operating unit.'))
+
+    @api.multi
+    @api.constrains('operating_unit_id')
+    def _check_required_operating_unit(self):
+        for rec in self:
+            if (rec.usage not in ('supplier', 'customer') and not
+                    rec.operating_unit_id):
+                raise UserError(
+                    _('Configuration error. The operating unit should be '
+                      'assigned to internal locations and to non other.')
```

Comment:
```suggestion
                      'assigned to internal locations only.')
```

Or rephrase because currently not easy to understand

Reply:
Fixed with your suggeestion